sqlglot/dialects/snowflake.py

```diff
@@ -267,6 +267,21 @@ class Parser(parser.Parser):
             ),
         }
 
+        def _parse_id_var(
+            self,
+            any_token: bool = True,
+            tokens: t.Optional[t.Collection[TokenType]] = None,
+        ) -> t.Optional[exp.Expression]:
+            if self._match_text_seq("IDENTIFIER", "("):
+                identifier = (
+                    super()._parse_id_var(any_token=any_token, tokens=tokens)
+                    or self._parse_string()
+                )
+                self._match_r_paren()
+                return self.expression(exp.ToIdentifier, this=identifier)
```

Comment:
This is reached when parsing DDL table names, since `schema` is set to `True` in that case and so we don't hit the `_parse_function` call here:

https://github.com/tobymao/sqlglot/blob/19295cc5d0080883af183771512f8e8a4050eecd/sqlglot/parser.py#L2311

In all other cases, `IDENTIFIER(...)` is parsed into `exp.ToIdentifier` in `_parse_function`.